pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -591,7 +591,7 @@ func verifySignature(connSignature *ConnectionSignature, recipientKeys string) (
 	// TODO: Replace with signed attachments issue-626
 	signatureSuite := ed25519signature2018.New()
 
-	err = signatureSuite.Verify(pubKey, sigData, signature)
+	err = signatureSuite.Verify("ed25519", pubKey, sigData, signature)
```

Comment:
I would prefer to have PublicKey struct that contains Type and Value. Both type and value are properties of public key. Later on if we need additional properties it will be easy to add. @Baha-sk

Reply:
I'm also working on constraining the key types as constants in the KMS, we can reference these types instead of open string value (once my PR is merged you can take advantage of it).

Reply:
Agree, @sandrask .  I will refactor that.